splink/blocking.py

```diff
@@ -48,6 +53,60 @@ def salted_blocking_rules(self):
             for n in range(self.salting_partitions):
                 yield f"{self.blocking_rule} and ceiling(l.__splink_salt * {self.salting_partitions}) = {n+1}"  # noqa: E501
 
+    @property
+    def _parsed_join_condition(self):
+        br = self.blocking_rule
+        return parse_one("INNER JOIN r", into=Join).on(
+            br, dialect=self.sqlglot_dialect
+        )  # using sqlglot==11.4.1
+
+    @property
+    def _join_conditions(self):
+        """
+        Extract the join conditions from the blocking rule as a tuple:
+        source_keys, join_keys, condition_expr
+
+        Returns:
+            list of tuples like [(name, name), (substr(name,1,2), substr(name,2,3))]
+        """
+
+        def remove_table_prefix(tree):
+            for c in tree.find_all(Column):
+                del c.args["table"]
+            return tree
+
+        j = self._parsed_join_condition
+
+        source_keys, join_keys, _ = join_condition(j)
+
+        keys = zip(source_keys, join_keys)
+
+        rmtp = remove_table_prefix
+
+        keys = [(rmtp(i), rmtp(j)) for (i, j) in keys]
+
+        keys = [
+            (i.sql(dialect=self.sqlglot_dialect), j.sql(self.sqlglot_dialect))
+            for (i, j) in keys
+        ]
+
+        keys = list(set(keys))
+
+        return keys
```

Comment:
You may well have noticed, but sqlglot also takes care of duplicated equalities... i.e.:
`"l.first_name = r.first_name"` yields the same `_join_conditions` as `"l.first_name = r.first_name and l.first_name = r.first_name"`

Reply:
Nice, I had not noticed

Reply:
I [deleted it](https://github.com/moj-analytical-services/splink/pull/1388/commits/e8f3f3c71d6d7cccb4b5b48f9eeda2af52e6c57f)